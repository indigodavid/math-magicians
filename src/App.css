*,
*::after,
*::before {
  box-sizing: border-box;
}

body {
  font-size: 1rem;
}

ul {
  list-style: none;
  padding: 0;
  margin: 0;
}

#calculator {
  display: flex;
  flex-direction: column;
  width: 30em;
  padding: 2px;
  background-color: #ccc;
}

#calculator-buttons {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 2px;
  text-align: center;
}

#input-display {
  appearance: none;
  font-size: 1rem;
  color: #fff;
  background-color: transparent;
  border: none;
  text-align: right;
  width: 100%;
}

#input-display:focus {
  border: none;
  outline: none;
}

.calculator-button {
  background-color: #ddd;
  color: #000;
  padding: 1rem;
  appearance: none;
  border: none;
  font-size: 1rem;
  margin: 0;
}

.calculator-button.orange {
  background-color: orange;
}

.button-zero {
  grid-column-start: 1;
  grid-column-end: 3;
}

.display {
  display: flex;
  justify-content: flex-end;
  text-align: right;
  background-color: #555;
  color: #fff;
  padding: 1rem 0.5rem;
  margin-bottom: 2px;
}
