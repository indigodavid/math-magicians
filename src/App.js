import './App.css';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Calculator from './components/Calculator';
import Home from './pages/Home';
import Quote from './pages/Quote';
import Navbar from './components/Navbar';
import NoMatch from './pages/NoMatch';

function App() {
  return (
    <Router>
      <Navbar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/calculator" element={<Calculator />} />
        <Route path="/quote" element={<Quote />} />
        <Route path="*" element={<NoMatch />} />
      </Routes>
    </Router>
  );
}

export default App;
