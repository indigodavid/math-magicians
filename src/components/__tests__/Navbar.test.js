import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../../App';

describe('Tests for Navbar', () => {
  test('Logo is displayed', () => {
    const { container } = render(<App />);
    const logo = container.querySelector('.logo');
    expect(logo).toBeInTheDocument();
  });

  test('Navbar links browsing', async () => {
    const { container } = render(<App />);

    expect(screen.getByText(/Welcome/i)).toBeInTheDocument();

    fireEvent.click(screen.getByText(/Quote/i));
    expect(container.querySelector('.quote')).toBeInTheDocument();
  });
});
