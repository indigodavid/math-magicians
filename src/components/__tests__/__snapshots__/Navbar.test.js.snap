// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Navbar Logo is displayed 1`] = `
<nav
  class="navbar"
>
  <div
    class="logo"
  >
    <a
      href="/"
    >
      Math Magicians
    </a>
  </div>
  <ul
    class="navLinks"
  >
    <li>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class=""
        href="/calculator"
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        class=""
        href="/quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
